now.go

```diff
@@ -1,9 +1,29 @@
 package utc
 
-import "time"
+import (
+	"sync"
+	"time"
+)
 
-// nowFn is the function used to get the current time and can be mocked with MockNow/MockNowFn
-var nowFn = now
+// nowFnMu protects access to nowFn
+var nowFnMu sync.Mutex
+
+func init() {
+	nowFn = now
+}
+
+// allowClock allows using a custom Clock and is called during test whenever setClock is called.
+func allowClock() {
```

Comment:
Please document why we do this whole dance with `allowClock`, `setClock` and `getClock`. Unless you know the reason - preventing race conditions with "mocked Now" and not incurring the cost of an atomic pointer in non-test code - it's very hard to figure out why the code was written like this.